### PR TITLE
Use appropriate return compounding

### PR DESCRIPTION
**Pull request by bhutraaditya**
Issue: #166 

For return series `r = (r1, r2, r3)` and `frequency=1`,

Current compounded return = `(r1 + r2 + r3) / 3`

Proposed compounded return = `(1+r1)(1+r2)(1+r3) ^ (1/3) - 1`

UTs are sensitive to the changes. One UT has been removed completely because of irrelevance.

### CODE REVIEW

File: pypfopt/expected_returns.py [hunk 1]
```diff
@@ -185,7 +185,7 @@ def james_stein_shrinkage(prices, returns_data=False, compounding=False, frequen
     :type prices: pd.DataFrame
     :param returns_data: if true, the first argument is returns instead of prices.
     :type returns_data: bool, defaults to False.
-    :param compounded: whether to properly compound the returns, optional.
+    :param compounding: whether to properly compound the returns, optional.
```
Comment (bhutraaditya) [inline pypfopt/expected_returns.py:188]: fixed typo

File: tests/test_expected_returns.py [hunk 2]
```diff
@@ -55,19 +55,14 @@ def test_mean_historical_returns_dummy():
         ]
     )
     mean = expected_returns.mean_historical_return(data, frequency=1)
-    test_answer = pd.Series([0.00865598, 0.025, 0.01286968, -0.03632333])
+    test_answer = pd.Series([0.0086559, 0.0250000, 0.0128696, -0.03632332])
     pd.testing.assert_series_equal(mean, test_answer)
+
     mean = expected_returns.mean_historical_return(data, compounding=True, frequency=1)
+    test_answer = pd.Series([0.0061922, 0.0241136, 0.0122722, -0.0421775])
```
Comment (bhutraaditya) [inline tests/test_expected_returns.py:62]: different result if the `compounding` switch is `True`

File: tests/test_expected_returns.py [hunk 3]
```diff
@@ -55,19 +55,14 @@ def test_mean_historical_returns_dummy():
...
     pd.testing.assert_series_equal(mean, test_answer)
 
 
-def test_mean_historical_returns_compounding():
-    df = get_data()
-    mean = expected_returns.mean_historical_return(df)
-    mean2 = expected_returns.mean_historical_return(df, compounding=True)
-    assert (mean2 >= mean).all()
-
```
Comment (bhutraaditya) [inline tests/test_expected_returns.py:69]: This assertion does not hold anymore

**Comment by robertmartin8** [comment]
@bhutraaditya 

Thanks for the suggested features. I need to think a bit more about how I want to implement this, because I want it to be robust to log returns also. Probably will need to have another parameter `log_returns=True` and an if statement.

Cheers,
Robert

**Pull request merged**